client/finality-grandpa/src/finality_proof.rs

```diff
@@ -148,8 +151,82 @@ impl<Block: BlockT> AuthoritySetForFinalityChecker<Block> for Arc<dyn FetchCheck
 	}
 }
 
+/// Justification for a finalized block.
+#[derive(Clone, Serialize, Deserialize)]
+pub struct JustificationNotification<Block: BlockT> {
+	/// Highest finalized block header
+	pub header: Block::Header,
+	/// An encoded justification proving that the given header has been finalized
+	pub justification: Vec<u8>,
```

Comment:
Isn't there a typedef for that?
Since we implement `Serialize/Deserialize` we should at least encode this as bytes, not a vector of numbers (default). Please add Serialization/deserialization test so that we can see how it's represented as string.

Reply:
This should be using the proper justification type https://github.com/paritytech/substrate/blob/master/client/finality-grandpa/src/justification.rs#L41 and not an encoded blob.

Reply:
I've changed this to use `GrandpaJustification` and not encoding it in `finalize_block` so that we don't use the encoded blob. 
What's the best way to deal with the serialization that happens in the rpc handler? Should we encode + serialize there?